### PR TITLE
kuliya: new recipe

### LINKED ISSUE #23527: [request] kuliya/1.0.0
### Package Name/Version

kuliya/1.0.0

### Webpage

https://dzcode-io.github.io/kuliya/

### Source code

https://github.com/dzcode-io/kuliya/tree/main/c

### Description of the library/tool

Algeria's college hierarchy dataset as a C library

### PR DESCRIPTION
**Pull request by omdxp**
Specify library name and version:  **kuliya/1.0.0**

Algeria's college hierarchy dataset as a C library

Fixes #23527 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/kuliya/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import os
+
+required_conan_version = ">=1.50.0"
+
+
+class JsmnConan(ConanFile):
+    name = "kuliya"
+    description = "Algeria's college hierarchy dataset as a C library."
+    license = "MIT"
+    topics = ("kuliya", "dataset", "api", "dz")
```
Comment (RubenRBS) [inline recipes/kuliya/all/conanfile.py:13]:
```suggestion
    topics = ("dataset", "api", "dz")
```

File: recipes/kuliya/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    homepage = "https://dzcode-io.github.io/kuliya/"
```
Comment (RubenRBS) [inline recipes/kuliya/all/conanfile.py:14]:
```suggestion
    homepage = "https://github.com/dzcode-io/kuliya"
```

This field's used in some npts that help to autimatically publish new verisons if the homepage points to the repo itself :)

File: recipes/kuliya/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        pass
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "kuliya.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include"))
```
Comment (RubenRBS) [inline recipes/kuliya/all/conanfile.py:34]: Does the library not need the `data.h` file too? Or am I missing something?

**Review by RubenRBS — COMMENTED** [review]
Hi! Thanks a lot! This looks great for the most part, only minor comments.

Don't worry about the CI failing, this will automatically get built once the CI systems are back online and it approves you :)

**Review by RubenRBS — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Comment by omdxp** [comment]
Adding the missing license file [here](https://github.com/dzcode-io/kuliya/pull/90)

**Comment by RubenRBS** [comment]
Thanks! if the code is not licensed separatedly from the whole repo, we can also use the license from the parent repository, let me know if that works better, and I can submit the code directly to this recipe to help you out :)

**Comment by RubenRBS** [comment]
Ok, I'll push some changes to your branch to acomplish this, give me a moment please!

**Comment by RubenRBS** [comment]
@omdxp just a heads-up:

When contributing to projects and creating a fork of the repository, it's better if you create a different branch on your fork on which to add your changes, and then open the PR from that branch (Instead of the current `master` you opened from) - It's easier for you in the future should you need to sync your fork without losing your work! :)

But it's not an issue for this PR, no worries, just something I just realized

**Comment by omdxp** [comment]
Yeah, will definitely keep this in mind, thanks for the heads up :)

File: recipes/kuliya/all/test_package/test_package.c [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#include <kuliya.h>
+
+int main()
+{
+    kuliya_init();
+    kuliya_schema *res = get_node_by_path("umkb/fst/dee/sec");
```
Comment (RubenRBS) [inline recipes/kuliya/all/test_package/test_package.c:6]: Are these files from the `_data` folder meant to be packaged with the library for it to work? Maybe we should be checking the output of the function somehow?

File: recipes/kuliya/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class JsmnConan(ConanFile):
+    name = "kuliya"
+    description = "Algeria's college hierarchy dataset as a C library."
+    license = "MIT"
+    topics = ("dataset", "api", "dz", "header-only")
+    homepage = "https://github.com/dzcode-io/kuliya"
+    url = "https://github.com/conan-io/conan-center-index"
+    package_type = "header-library"
```
Comment (RubenRBS) [inline recipes/kuliya/all/conanfile.py:16]: Upstream's CMakeLists seems to imply it wants to compile this library against a few other ones as dependencies, but those do not show up here. Are those just for educational pourposes and are they not needed in reallity? Or does the library need them?

**Review by RubenRBS — COMMENTED** [review]
Looking great now! I have a few minor comments, but this looks nice, thanks a lot for taking the time to submit the PR

Some more notes looking into upstream: The ideal thing would be for the license to be bundled in the release tar. Maybe your github action can copy the parent LICENSE file into the output folder?

File: recipes/kuliya/all/test_package/test_package.c (hunk 4 above)
Comment (omdxp) [reply recipes/kuliya/all/test_package/test_package.c:6]: there is a `build.c` script to generate `data.h` file from that `_data` folder, so these are not read from that folder but instead they are static variables identified by these paths

File: recipes/kuliya/all/conanfile.py (hunk 5 above)
Comment (omdxp) [reply recipes/kuliya/all/conanfile.py:16]: This recipe has no dependencies, those other dependencies are just needed for the `build.c` which is intended to be used only on the project actions.

TL;DR
The `kuliya` recipe will only have `kuliya.h` and `data.h` files

**Comment by omdxp** [comment]
> Some more notes looking into upstream: The ideal thing would be for the license to be bundled in the release tar. Maybe your github action can copy the parent LICENSE file into the output folder?

Will apply this for the next release whenever we have one

File: recipes/kuliya/all/test_package/test_package.c (hunk 4 above)
Comment (omdxp) [reply recipes/kuliya/all/test_package/test_package.c:6]: I will add assertion for the output

**Review by RubenRBS — APPROVED** [review]
Thanks a lot for the extra insight! :)

**Review by toge — APPROVED** [review]
